deepchem/feat/binding_pocket_features.py

```diff
@@ -1,18 +1,61 @@
 """
 Featurizes proposed binding pockets.
 """
-__author__ = "Bharath Ramsundar"
-__copyright__ = "Copyright 2017, Stanford University"
-__license__ = "MIT"
-
 import numpy as np
-from deepchem.utils.save import log
+import logging
+from deepchem.utils import rdkit_util
 from deepchem.feat import Featurizer
 
+logger = logging.getLogger(__name__)
 
-class BindingPocketFeaturizer(Featurizer):
+
+def boxes_to_atoms(coords, boxes):
+  """Maps each box to a list of atoms in that box.
+
+  Given the coordinates of a macromolecule, and a collection of boxes,
+  returns a dictionary which maps boxes to the atom indices of the
+  atoms in them.
+
+  Parameters
+  ----------
+  coords: np.ndarray
+    Of shape `(N, 3)
+  boxes: list
+    list of `CoordinateBox` objects.
+
+  Returns
+  -------
+  dictionary mapping `CoordinateBox` objects to lists of atom coordinates
   """
-  Featurizes binding pockets with information about chemical environments.
+  mapping = {}
+  for box_ind, box in enumerate(boxes):
+    box_atoms = []
+    for atom_ind in range(len(coords)):
+      atom = coords[atom_ind]
+      if atom in box:
+        box_atoms.append(atom_ind)
+    mapping[box] = box_atoms
```

Comment:
If the number of atoms is large, and if this turns out to be a bottleneck, there are much faster ways of doing it.

Reply:
Would you have suggestions for better algorithms here? It would be useful to have this technique be fast in cast we want to use it at large scale.

Reply:
This code scales as O(n*m) where n is the number of atoms and m is the number of boxes.  If n and m are both large, a faster approach is to first assign atoms to voxels, which takes a single loop of O(n), and then for each box loop over the voxels that intersect it, which scales as O(m).

It's probably best to leave this code as it is for the moment, then see how it works in practice and whether it needs to be optimized.